### ISSUE TITLE
Policy group support

### ISSUE THREAD

**Issue by zedtux** [EugenMayer/chef-tinc-cookbook#24]
This cookbook lack support of policy [groups/environments](https://docs.chef.io/environments/) (should be the same for role/environments).

Given the following Policyfile:

```ruby
name 'test'
cookbook 'tincvpn', '~> 1.0.0'

default['env-1'] = {
  tincvpn: {
    ...
  }
}

default['env-2'] = {
  tincvpn: {
    ...
  }
}
```

This cookbook ignores them and takes the default attributes.

The fix is about to check the `node.environment` and:

* when it is `_default`, use the existing way to access attributes
* when it is different than `_default`, use `node[node.environment]` to access attributes instead

But it is also to update the way to store the data.

I'm preparing a PR I'll push soon.

**Comment by EugenMayer** [comment]
Will this still be compatible using `knife-zero` then? ( that would be a requirement for me ). What i mean is, if we add policfile support, does this block `knife-zero` deployments?

**Comment by zedtux** [comment]
Actually I tried to mean that by "should be the same for role/environments", sorry for the poor wording.

So today you should already have the issue that the cookbook is not taking into account the environment from the role/environment model, nor the group from the policy/group model.

As an example, the `search` is not filtering on the environment/group. We are doing the following:

```ruby
peers = search(:node, "tincvpn_networks_#{network_name}_host_pubkey:*")
```

Which would have to be something like that based on [my understanding of the documentation](https://docs.chef.io/chef_search/#environments):

```ruby
peers = search(:node, "tincvpn_networks_#{network_name}_host_pubkey:* AND chef_environment:#{node.environment}")
```

> Will this still be compatible using knife-zero then? ( that would be a requirement for me ).

For me too, I'm using knife zero to provision my nodes, so we are in the same boat :wink:.

**Comment by zedtux** [comment]
This issue can be closed since the PR #28 has been merged.
